Modules/xSharePoint/DSCResources/MSFT_xSPPasswordChangeSettings/MSFT_xSPPasswordChangeSettings.psm1

```diff
@@ -0,0 +1,85 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)][System.String] $MailAddress,
+        [parameter(Mandatory = $false)][ValidateRange(0,356)][System.UInt32] $DaysBeforeExpiry,
+        [parameter(Mandatory = $false)][ValidateRange(0,36000)][System.UInt32] $PasswordChangeWaitTimeSeconds,
+        [parameter(Mandatory = $false)][ValidateRange(0,99)][System.UInt32] $NumberOfRetries
+    )
+
+    Write-Verbose -Message "Retrieving farm wide automatic password change settings"
+
+    $result = Invoke-xSharePointCommand -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        $farm = Get-SPFarm
+        if ($null -eq $farm ) { return $null }
+        return @{
+            MailAddress = $farm.PasswordChangeEmailAddress
+            PasswordChangeWaitTimeSeconds= $farm.PasswordChangeGuardTime
+            NumberOfRetries= $farm.PasswordChangeMaximumTries
+            DaysBeforeExpiry = $farm.DaysBeforePasswordExpirationToSendEmail 
+        }
+    }
+    return $result
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)][System.String] $MailAddress,
+        [parameter(Mandatory = $false)][ValidateRange(0,356 )][System.UInt32] $DaysBeforeExpiry,
```

Comment:
according to the UI:(http://spfarm-sp:8080/_admin/PasswordSettings.aspx)

The value for days before expiry is invalid or out of range. The maximum
value is 356 days.

On Sun, Nov 15, 2015 at 10:54 PM, Brian Farnhill <notifications@github.com>
wrote:

> In
> Modules/xSharePoint/DSCResources/MSFT_xSPPasswordChangeSettings/MSFT_xSPPasswordChangeSettings.psm1
> <https://github.com/PowerShell/xSharePoint/pull/95#discussion_r44870806>:
>
> > +            MailAddress = $farm.PasswordChangeEmailAddress
> > +            PasswordChangeWaitTimeSeconds= $farm.PasswordChangeGuardTime
> > +            NumberOfRetries= $farm.PasswordChangeMaximumTries
> > +            DaysBeforeExpiry = $farm.DaysBeforePasswordExpirationToSendEmail
> > +        }
> > +    }
> > +    return $result
> > +}
> > +
> > +function Set-TargetResource
> > +{
> > +    [CmdletBinding()]
> > +    param
> > +    (
> > +        [parameter(Mandatory = $true)][System.String] $MailAddress,
> > +        [parameter(Mandatory = $false)][ValidateRange(0,356 )][System.UInt32] $DaysBeforeExpiry,
>
> Is this meant to be 365 not 356?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PowerShell/xSharePoint/pull/95/files#r44870806>.
>